### ISSUE TITLE
Linking issue in initial "make" (posDaemon)

### ISSUE THREAD

**Issue by Nazdravi** [legatoproject/legato-af#55]
Hi,
in the attempt to generate the legato framework (downloaded with "repo") the "make"-command runs for a while and fails on linking the "posDaemon" shared library:
```
[832/884] Linking C executable
FAILED: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/app/positioningService/obj/posDaemon/posDaemon 
/usr/bin//gcc  -Wl,--build-id -g -fPIE -pie -o /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/app/positioningService/obj/posDaemon/posDaemon /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/app/positioningService/obj/posDaemon/_main.c.o -rdynamic -Wl,--enable-new-dtags,-rpath="\$ORIGIN/../lib:$LEGATO_BUILD/framework/lib" -L "-L/home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj" -lComponent_posDaemon "-L/home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/6a4e06096b70caf6262507a17c9b7bce/obj" -lComponent_watchdogChain "-L/home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj" -lComponent_posDaemon "-L/home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/6a4e06096b70caf6262507a17c9b7bce/obj" -lComponent_watchdogChain "-L$LEGATO_BUILD/framework/lib" -llegato -lpthread -lrt -ldl -lm  -g && splitdebug -d /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/debug /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/app/positioningService/obj/posDaemon/posDaemon
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_DisableExternalLna'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetNmeaSentences'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_DisableExtendedEphemerisFile'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_EnableExtendedEphemerisFile'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetMaxNmeaRate'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetSupportedNmeaSentences'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_EnableExternalLna'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_InjectSuplCertificate'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_DeleteSuplCertificate'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_LoadExtendedEphemerisFile'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_AddNmeaHandler'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_Stop'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_ForceEngineStop'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_Disable'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_SetConstellation'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_SetConstellationArea'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_SetSuplAssistedMode'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetConstellationArea'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetConstellation'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetExternalLna'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_InjectUtcTime'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetExtendedEphemerisValidity'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_SetNmeaSentences'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetMinNmeaRate'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_AddPositionDataHandler'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetTtff'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_DeleteAssistData'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_Enable'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_ConvertDataCoordinateSystem'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetSupportedConstellations'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_Init'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_RemovePositionDataHandler'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetMinElevation'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_Start'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetSuplAssistedMode'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetAcquisitionRate'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_GetLeapSeconds'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_SetSuplServerUrl'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_SetMinElevation'
/usr/bin/ld: /home/arndt/Entwicklung/MangOH/legato-framework/legato/build/localhost/system/component/56f12bdd07075a7fd95b5c454bbf489d/obj/libComponent_posDaemon.so: undefined reference to `pa_gnss_SetAcquisitionRate'
collect2: Fehler: ld gab 1 als Ende-Status zurück
[837/884] Compiling C source
ninja: build stopped: subcommand failed.
make[1]: *** [Makefile:542: system] Fehler 1
make[1]: Verzeichnis „/home/arndt/Entwicklung/MangOH/legato-framework/legato“ wird verlassen
make: *** [Makefile:460: default] Fehler 2
```
Did anyone already experience this and there is eventually a quick fix?
Kind regards and nazdravi

**Comment by CoRfr** [comment]
Just making sure, how did you clone Legato ?

**Comment by Nazdravi** [comment]
Legato was cloned with:
```
#> repo init -u git://github.com/legatoproject/manifest &&  repo sync
#> cd legato
#> make
```
(I believe this is the (one of) recommended sequence)

**Comment by CoRfr** [comment]
If you do `make localhost`, does that work?
The format is `make <target>` where target is localhost, wp76xx, wp85, ...

**Comment by Nazdravi** [comment]
No change with "make localhost" - same error at the same stage of compilation.

**Comment by CoRfr** [comment]
What is your OS? Ubuntu 18.04 ?

**Comment by Nazdravi** [comment]
Arch Linux (current):
```
automake 1.16.2-1
cmake 3.17.1-2
gcc 9.3.0-1
make 4.3-3
repo 2.5-1
```

**Comment by CoRfr** [comment]
Right, the build has been failing for sometimes on Arch I think. Usually it's because of the compiler being much more recent than the Ubuntu LTS that we officially support as a dev env.
I also noticed this particular issue 2 months ago but it has not been fixed so far.

I have a potential fix: https://github.com/CoRfr/legato-af/commit/8309dd28392f302ee952b3d127bc20df2bd41a4a
I tested it locally and it builds.
